lib/matplotlib/axes/_axes.py

```diff
@@ -1352,8 +1361,9 @@ def eventplot(self, positions, orientation='horizontal', lineoffsets=1,
         if len(linewidths) == 1:
             linewidths = np.tile(linewidths, len(positions))
         if len(colors) == 1:
-            colors = list(colors)
-            colors = colors * len(positions)
+            colors = list(colors) * len(positions)
+        if len(alpha) == 1:
+            alpha = list(alpha) * len(positions)
```

Comment:
You don't need the `list` here? The one on `colors` is because it _may_ be passed through `to_rgba_array` to become a NumPy array instead, but this is still a `list`, so this just makes an extra copy.
```suggestion
            alpha = alpha * len(positions)
```

Reply:
I think this would fail for `alpha=np.array([0.5])`. Admittedly it's a bit constructed, but I can imagine generic code that creates such 1-element arrays. `np.tile` would be the alternative.

Reply:
I suppose that's true; seems fine with the `list` then.